portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppHelperLocalServiceImpl.java

```diff
@@ -1103,9 +1103,11 @@ public void updateStatus(
 					fileEntry.getFileEntryId());
 
 				if (assetEntry != null) {
-					assetEntryLocalService.updateVisible(
+					assetEntryLocalService.updateEntry(
```

Comment:
Hey @crodriguezy instead of doing the bug fix here, I thinnk that the problem is here https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppHelperLocalServiceImpl.java#L885 and https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppHelperLocalServiceImpl.java#L949

We should always use the publishDate no matter if the document is visible or not. I think if we remove those if conditionals it should fix this issue.

Can you please also create an integration test for this to ensure that it works?

Let me know if you have any question.

Thanks!

Reply:
Hi @sergiogonzalez , I just realized, what if  *dl.file.entry.drafts.enabled* property is set to **true** (so draft documents could be created and published/saved later)? The solution you proposed will set a not null publishDate no matter it is published or not. 

On the other hand, JournalArticle assetEntries' publishDate is set this way: if it is in draft state, publishDate is null. Otherwise, publishDate is set to createDate.

I think that to be consistent with JournalArticle behaviour we must follow the same process for Documents. 

What do you think? If you agree with that, I will send you the same PR rebased, and if there is no problem I will send you a second one later including the integration test, because a customer is demanding a patch for this and the creation of the test for it will take me some extra time.

Thanks!

Reply:
Hi @sergiogonzalez , I finally sent you a new PR with the same change as this one and a modification for one of the Persistence test, related to the publish date.

Can you please review it? The customer is waiting for this solution and some other fixes are pending on the resolution of thiin order to be  included together in a hotfix.

https://github.com/sergiogonzalez/liferay-portal/pull/4351

Thanks a lot!